File: component_gank/src/main/java/com/baiiu/gank/GankListFragment.java
package com.baiiu.gank;

import com.baiiu.common.base.BaseFragment;

/**
 * author: baiiu
 * date: on 17/10/27 11:18
 * description:
 */
public class GankListFragment extends BaseFragment {

    public static GankListFragment instance() {
        return new GankListFragment();
    }

    @Override public int provideLayoutId() {
        return R.layout.fragment_gank;
    }
}

File: component_gank/src/main/java/com/baiiu/gank/GankListFragment.kt
package com.baiiu.gank

import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.baiiu.common.base.BaseFragment
import kotlinx.android.synthetic.main.gank_fragment_gank.*

/**
 * author: zhuzhe
 * time: 2020-01-08
 * description:
 */
class GankListFragment : BaseFragment() {

    val viewModel by lazy {
        ViewModelProviders.of(this).get(SimpleModel::class.java)
    }

    override fun provideLayoutId(): Int {
        return R.layout.gank_fragment_gank
    }


    override fun initOnCreateView() {
        super.initOnCreateView()

        viewModel.start()

        viewModel.strLiveData.observe(this, Observer {
            it?.also {

                android.util.Log.e("mLogU", "observe: $it")

                textView.text = it
            }
        })
    }

}